http/http_client.hpp

```diff
@@ -468,7 +468,7 @@ class ConnectionInfo : public std::enable_shared_from_this<ConnectionInfo>
             }
             if (connPolicy->retryPolicyAction == "RetryForever")
             {
-                state = ConnState::idle;
+                state = ConnState::closed;
```

Comment:
Can you explain in the PR message, why we need this?

Reply:
we reach here when we are not able to send data to hmc due connection error. we must recreate new socket before attempting to try again. close will make sure that the next try is with new socket else it will attempt with old socket and evnetually fails. We can avoid this failure if we make state closed

Reply:
I updated the title and crossed some text out of the PR message. For next time, please make sure your PR msg, etc reflects the latest state.